### PR TITLE
Add unique identifier to Person class

### PR DESCRIPTION
**Pull request by hhdqirui**
Add unique identifier to Person class. The field `id` is included to be used for identifying a unique person.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -107,10 +107,11 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
                 .orElse(personToEdit.getQuarantineStatus());
         InfectionStatus updatedInfectionStatus = editPersonDescriptor.getInfectionStatus()
                                             .orElse(personToEdit.getInfectionStatus());
+        Index updatedId = editPersonDescriptor.getId().orElse(personToEdit.getId());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
 
         return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress,
-                updatedQuarantineStatus, updatedInfectionStatus, updatedTags);
+                updatedQuarantineStatus, updatedInfectionStatus, updatedId, updatedTags);
```
Comment (hopinxian) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:114]: Ids are abstracted away from the user. Users should not be aware of Ids, and thus should not be given the opportunity to change the id of a person. I feel that thus editpersondescriptor should not have an id field and id is automatically retrieved from the person that is being edited.

File: src/test/java/seedu/address/model/person/UniquePersonListTest.java [hunk 2]
```diff
@@ -47,6 +50,31 @@ public void contains_personWithSameIdentityFieldsInList_returnsTrue() {
...
+    }
+
+    @Test
+    public void containsSameIdPerson_sameIdDifferentIdentity_returnsTrue() {
+        uniquePersonList.add(ALICE);
+        Person editedPerson = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
+                .build();
+        assertTrue(uniquePersonList.containsSameIdPerson(editedPerson));
+    }
+
```
Comment (hopinxian) [inline src/test/java/seedu/address/model/person/UniquePersonListTest.java:77]: Some additional tests can be conducted in this class. Such as tests for adding unidentifiable locations and tests for setting location lists which dont fulfill the identifiable property. You may refer to #118 UniqueLocationListTest for reference.

**Review by hopinxian — CHANGES_REQUESTED** [review]
A few comments are made. After finishing them, this can be approved.

File: src/main/java/seedu/address/model/person/UniquePersonList.java [hunk 3]
```diff
@@ -65,6 +77,8 @@ public void setPerson(Person target, Person editedPerson) {
             throw new DuplicatePersonException();
         }
 
+        // assert(target.isSameId(editedPerson));
+
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/seedu/address/model/person/UniquePersonList.java:81]: Fixed

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (hhdqirui) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:114]: Fixed.

File: src/test/java/seedu/address/model/person/UniquePersonListTest.java (hunk 2 above)
Comment (hhdqirui) [reply src/test/java/seedu/address/model/person/UniquePersonListTest.java:82]: More tests added.

Comment (hopinxian) [reply src/test/java/seedu/address/model/person/UniquePersonListTest.java:82]: One more test was missed.

**Review by KohHanMing — APPROVED** [review]
Looks good!

File: src/test/java/seedu/address/model/person/UniquePersonListTest.java (hunk 2 above)
Comment (hhdqirui) [reply src/test/java/seedu/address/model/person/UniquePersonListTest.java:89]: Test added.

**Review by hopinxian — APPROVED** [review]
LGTM

**Pull request merged**